Lib/asyncio/base_events.py

```diff
@@ -856,6 +797,15 @@ async def getaddrinfo(self, host, port, *,
         else:
             getaddr_func = socket.getaddrinfo
 
+        try:
+            return getaddr_func(
```

Comment:
this is inspired by the implementation from trio https://github.com/python-trio/trio/blob/4edfd41bd5519a2e626e87f6c6ca9fb32b90a6f4/trio/_socket.py#L125-L192